cmd/kubeadm/app/cmd/init.go

```diff
@@ -336,6 +337,9 @@ func newInitData(cmd *cobra.Command, args []string, options *initOptions, out io
 		return nil, err
 	}
 
+	// For new clusters we want to set the kubelet cgroup driver to "systemd" unless the user is explicit about it
+	componentconfigs.MutateCgroupDriver(&cfg.ClusterConfiguration)
```

Comment:
For sake of future memory, I will add a note:
```
 // Currently this cannot be as part of Default() because the function is called for
 // upgrades too, which can break existing nodes after a kubelet restart.
```